README.md

```diff
@@ -1,28 +1,142 @@
 # IOS Vast Client
 
-IOS Vast Client is a Swift Framework which implements the [VAST 3.0](https://www.iab.com/guidelines/digital-video-ad-serving-template-vast-3-0/) spec.
+IOS Vast Client is a Swift Framework which implements the [VAST 4.0](https://www.iab.com/guidelines/digital-video-ad-serving-template-vast/) spec.
 
 This project is ongoing and in very early stages of development. As this project is being developed we will only target a subset of the VAST spec to meet our immediate needs. We do not recommend to use this library at this time. 
 
 ## Features
 
-* VAST 3.0 Spec Complient
+* VAST 4.0 Spec Complient
 * Vast XML Parser and Validator
 * VAST Impression and Ad Tracking
 * VAST Error Tracking
 
 ## API docs
 
-TODO
+iOS Vast Client contains 2 interfaces for public use.
+
+### VastClient
+
+Fetch and parse VAST file from URL. You can customize behaviour with `VastClientOptions` during initialization of `VastClient`.
+
+`VastClient` result is `VastModel` structure or `Error` in case the fetch failed. You should pass this model to `VastTracker`.
+
+### VastTracker
+
+Handels tracking of ad break progress and errors. Notifies state updates via delegate calls.
+
+#### Initialization
+
+Init `VastTracker` with `VastModel` structure but always make sure to use the actual `VastAd` and other information provided via delegate function calls - do not keep the `VastModel` as not all information from the `VastModel` might be valid for playback etc.
+
+If you initialize `VastTracker` with delegate - `func adBreakStart(vastTracker: VastTracker, totalAds: Int)` will be called immediately. 
+
+#### Progress
+
+`func updateProgress(time: Double) throws`
+After you initialize `VastTracker` you should call this function with `time = 0` parameter to start tracking ad playback. `VastTracker` will select linear ad to play and will notify you via delegate call `func adStart(vastTracker: VastTracker, ad: VastAd)`
```

Comment:
One of the desired purposes of the progress is for ads that are inline with content. So the player displaying the ad will most likely not have a playhead of 0 at the time of the ad. So maybe for the description we shouldn't explicitly state to always use time = 0.

Reply:
I realised an issue here on my side, that works for great for our usage at the moment but breaks when used for inline ads i believe.

I might need to understand the playhead a bit more to fix this.

I see a possible issue, that might result in ad start being tracked pre-maturely and ads starting at wrong times.

Would you maybe be open for a deeper discussion? I will post my current fixes and we can take it from there.

Reply:
The main thing is:

When playing 2 pre-roll inline ads, how the progress should be updated to work for your case?

Ad 1: duration 15 seconds
Ad 2: duration 20 seconds, skippable after 5 seconds of playback

Content in length of 5 minutes.


Init VastTracker with `startTime = 0`
call `updateProgress(time: 0)`
Ad 1 will start playing
Ad 1 finishes in time 15s
Ad 2 starts

Q1: What time should the client send to update progress? 0 or 15?

Ad 2 finishes by user skipping it in 5 seconds after playback starts

Q2: What time should we supply to `updateProgress(time:)` at this moment when normal content should start playing from 0?

How should this content update tracking work with multiple adBreaks in VMAP model?